ansible_collections/studyly/git_mactp/plugins/module_utils/git_actions.py

```diff
@@ -255,3 +305,36 @@ def push(self):
                 "git_push": {"output": str(output), "error": str(error), "changed": True}
             }
         FailingMessage(self.module, rc, command, output, error)
+
+    def tag(self):
+        """
+        Tags the current state of repo
+
+        args:
+            * module:
+                type: dict()
+                description: Ansible basic module utilities and module arguments.
+        return:
+            * result:
+                type: dict()
+                description: returned output from git tag command and updated changed status.
+        """
+        tag_name: str = self.module.params["tag"]
+        message: str = self.module.params["message"]
+
+        command = [self.git_path, "tag", "-am", (message if message else tag_name), tag_name]       
+
+        # TODO:
+        # add a better way of handeling tag messages
+        # - if message was not given, prompt for message
+        # - if message was given but is empty, don't use any
+        # - if message was given and is not empty, use this as message
+
+        # Push result is returned in stderr instead of stdout, hence vars position is inverted.
+        rc, error, output = self.module.run_command(command, cwd=self.path)
+        
+        if rc == 0:
+            return {
+                "git_tag": {"output": str(output), "error": str(error), "changed": True}
```

Comment:
Just a note: Assuming a change is valid in this case as it is forbidden to create a tag twice, thus git will fail instead of accepting with "nothing changed".